File: src/main/java/jp/artan/japanesefoodmod/common/potion/JapanesePotionType.java
package jp.artan.japanesefoodmod.common.potion;

import jp.artan.japanesefoodmod.JapaneseFoodMod;
import net.minecraft.potion.PotionType;

import static jp.artan.japanesefoodmod.init.Init.POTION_TYPES;

public class JapanesePotionType extends PotionType {
    private final String name;

    public JapanesePotionType(String name) {
        super();
        this.name = name;
        this.setRegistryName(JapaneseFoodMod.MODID, this.name);

        POTION_TYPES.add(this);
    }
}

File: src/main/java/jp/artan/japanesefoodmod/common/potion/JapanesePotionType.kt
package jp.artan.japanesefoodmod.common.potion

import jp.artan.japanesefoodmod.JapaneseFoodMod
import jp.artan.japanesefoodmod.init.Init.Companion.POTION_TYPES
import net.minecraft.potion.PotionType

class JapanesePotionType(private val potionTypeName: String) : PotionType() {

    init {
        this.setRegistryName(JapaneseFoodMod.MODID, this.potionTypeName)
        POTION_TYPES.add(this)
    }
}
